### PR TITLE
XWIKI-21412: Move the Tour Application to xwiki-platform

### PR DESCRIPTION
**Pull request by Sereza7**
**EDIT:** See https://github.com/xwiki/xwiki-platform/pull/2453#issuecomment-1755663113 for an updated PR starting message
______
______

## Issue
Fix for https://ge.xwiki.org/scans/tests?search.relativeStartTime=P28D&search.tags=master&search.timeZoneId=Europe%2FBerlin&tests.container=org.xwiki.tour.test.ui.TourApplicationIT&tests.test=verifyTourFeatures
This is a regression caused by the merge for [TOUR-81](https://jira.xwiki.org/browse/TOUR-81).
## PR Changes
* Updated the 'hidden' attribute of the button when opening/closing the tour modal
* Overrode the .btn 'display' property when this button is hidden.

## View
[Demo video](https://up1.xwikisas.com/#hxyRVwzGJ_h_172P8wWT-w)
We can see on this video that with the changes brought in this PR, the tour button disappears when the tour is opened, as expected by the tests.

### CODE REVIEW

**Comment by surli** [comment]
@Sereza7 I think I would use https://jira.xwiki.org/browse/XWIKI-21412 for this ticket, indicating in a comment the regression from TOUR-81 and linking the tickets. Someone going to TOUR-81 could find the info then. In general I'm not a big fan of bug fixes with a Misc personally...

**Comment by Sereza7** [comment]
## Jira
https://jira.xwiki.org/browse/XWIKI-21412
## PR Changes
* Updated the 'hidden' attribute of the button when opening/closing the tour modal
* Overrode the .btn 'display' property when this button is hidden.

## Notes
Fix for https://ge.xwiki.org/scans/tests?search.relativeStartTime=P28D&search.tags=master&search.timeZoneId=Europe%2FBerlin&tests.container=org.xwiki.tour.test.ui.TourApplicationIT&tests.test=verifyTourFeatures
This is a regression caused by the merge for [TOUR-81](https://jira.xwiki.org/browse/TOUR-81).
## View
[Demo video](https://up1.xwikisas.com/#hxyRVwzGJ_h_172P8wWT-w)
We can see on this video that with the changes brought in this PR, the tour button disappears when the tour is opened, as expected by the tests.

File: xwiki-platform-core/xwiki-platform-tour/xwiki-platform-tour-ui/src/main/resources/TourCode/TourJS.xml [hunk 1]
```diff
@@ -545,6 +550,9 @@ require(['jquery', 'xwiki-meta'], function ($, xm) {
 
   &amp; &gt; #tourResume {
     width: fit-content;
+    &amp;[hidden] {
+      display: none;
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-tour/xwiki-platform-tour-ui/src/main/resources/TourCode/TourJS.xml:554]: Is this really needed?

Comment (Sereza7) [reply xwiki-platform-core/xwiki-platform-tour/xwiki-platform-tour-ui/src/main/resources/TourCode/TourJS.xml:554]: Without this rule, the '.btn' display would take priority over the expected [hidden] behavior. This meant that adding the attribute 'hidden' to a .btn would not change the way it was displayed, which defeats the purpose of adding this attribute.

This change restablishes the way hidden is expected to act on this .btn .

**Pull request merged**